Add support for Rance 2 Hint Disk

kichikuou opened the pull request:
The `NACT_Sys1` changes were contributed by @RottenBlock.

To get this to work, you need to put the followings in the current directory:

- all the `.DAT` files for Rance 2,
- `GDISK.DAT`/`GCG.DAT` from the hint disk, and
- `AG00.DAT` from the hint disk renamed to `GG00.DAT`

And run system3 with the game id `rance2_hint`.

Remaining issue:
- The positions of message box and menu must be adjusted.

kichikuou committed:
RottenBlock's changes for Rance 2 Hint Disk

kichikuou committed:
Load GDISK.DAT and GG00.DAT if game_id is rance2_hint

AG00.DAT on the hint disk must be renamed to GG00.DAT to avoid conflicts
with AG00.DAT of the main game. This is the setup used in the Ryu1's
porting kit (http://www.ryu1.jp/alice/r2port.html).

kichikuou committed:
Clean up

kichikuou commented:
I'm only testing it on the Japanese version.

@RottenBlock Since I've made system3 read `GDISK.DAT` and `GG00.DAT`, your test environment may need to be adjusted.

RottenBlock commented:
Ah, my bad, I forgot to include ags.cpp in my zip, so the message and dialog boxes got left out. Here's what I originally put:

`		case CRC32_RANCE2:
		case CRC32_RANCE2_HINT:
			SET_TEXT(i, 8, 285, 502, 396, true);
			SET_MENU(i, 431, 19, 624, 181, true);
			break;`

kichikuou committed:
Set text/menu box locations for Rance 2

Contributed by @RottenBlock.

kichikuou commented:
Great, added it to this PR (I changed the last argument to `false` to not draw a frame around the menu).

Are there any concerns left, including about the code I added? If not, I will merge this and publish a new release.

kichikuou committed:
README: Add rance2 and rance2_hint to the game ID table

RottenBlock commented:
Oh, good thinking about the border. I forgot that was a thing. I tried changing the color to replicate the effect but the palette's cyan isn't the same as the PC-98's. Your solution's superior here, I think.

I don't have any other major concerns, I think you can run with it!

kichikuou commented:
Released as [v1.4.0](https://github.com/kichikuou/system3-sdl2/releases/tag/v1.4.0).
